src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/rca/RcaControllerHelper.java

```diff
@@ -0,0 +1,108 @@
+  public static final String CAT_MASTER_URL = "http://localhost:9200/_cat/master?h=ip";
+  private static String ELECTED_MASTER_RCA_CONF_PATH = RcaConsts.RCA_CONF_MASTER_PATH;
+  private static String MASTER_RCA_CONF_PATH = RcaConsts.RCA_CONF_IDLE_MASTER_PATH;
+  private static String RCA_CONF_PATH = RcaConsts.RCA_CONF_PATH;
+
+  /**
+   * Picks a configuration for RCA based on the node's role.
+   *
+   * @param nodeRole The role of the node(data/eligible master/elected master)
+   * @return The configuration based on the role.
+   */
+  public static RcaConf pickRcaConfForRole(final NodeRole nodeRole) {
+    if (NodeRole.ELECTED_MASTER == nodeRole) {
+      LOG.debug("picking elected master conf");
+      return new RcaConf(ELECTED_MASTER_RCA_CONF_PATH);
+    }
+
+    if (NodeRole.MASTER == nodeRole) {
+      LOG.debug("picking idle master conf");
+      return new RcaConf(MASTER_RCA_CONF_PATH);
+    }
+
+    if (NodeRole.DATA == nodeRole) {
+      LOG.debug("picking data node conf");
+      return new RcaConf(RCA_CONF_PATH);
+    }
+
+    LOG.debug("picking default conf");
+    return new RcaConf(RCA_CONF_PATH);
+  }
+
+  /**
+   * Gets the elected master's information by performing a _cat/master call.
+   *
+   * @return The host address of the elected master.
+   */
+  public static String getElectedMasterHostAddress() {
+    try {
+      LOG.info("Making _cat/master call");
+      PerformanceAnalyzerApp.RCA_RUNTIME_METRICS_AGGREGATOR.updateStat(
+          RcaRuntimeMetrics.ES_APIS_CALLED, "catMaster", 1);
+
+      final URL url = new URL(CAT_MASTER_URL);
+      HttpURLConnection connection = (HttpURLConnection) url.openConnection();
+      connection.setRequestMethod("GET");
+      BufferedReader in = new BufferedReader(new InputStreamReader(connection.getInputStream()));
+      String inputLine = in.readLine();
+      in.close();
+
+      return inputLine;
+    } catch (IOException e) {
+      LOG.error("Could not get the elected master node: {}", e.getMessage());
+      e.printStackTrace();
+    }
+
+    return "";
+  }
+
+  /**
+   * Builds a thread pool used by the networking layer to pass messages and perform networking
+   * functions.
+   *
+   * @param queueLength The length of the queue in the threadpool.
+   * @return The thread pool as an executor service.
+   */
+  public static ExecutorService buildNetworkThreadPool(final int queueLength) {
+    final ThreadFactory rcaNetThreadFactory =
+        new ThreadFactoryBuilder().setNameFormat(RcaConsts.RCA_NETWORK_THREAD_NAME_FORMAT)
+                                  .setDaemon(true)
+                                  .build();
+    final BlockingQueue<Runnable> threadPoolQueue = new LinkedBlockingQueue<>(queueLength);
+    return new ThreadPoolExecutor(RcaConsts.NETWORK_CORE_THREAD_COUNT,
+        RcaConsts.NETWORK_MAX_THREAD_COUNT, 0L, TimeUnit.MILLISECONDS, threadPoolQueue,
+        rcaNetThreadFactory);
```

Comment:
I was thinking that we move the entire thread creation into a helper class. The thread creators should use that to create threads using few parameters